extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -333,4 +336,15 @@ public static boolean writeResultsToDurableStorage(final MSQSpec querySpec)
   {
     return querySpec.getDestination() instanceof DurableStorageMSQDestination;
   }
+
+  @Override
+  public LookupLoadingSpec getLookupLoadingSpec()
+  {
+    if (getQuerySpec().getQuery().getContext().containsKey(PlannerContext.CTX_LOOKUPS_TO_LOAD)) {
+      List<String> lookupsToLoad = (List<String>) getQuerySpec().getQuery().getContext().get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
```

Comment:
I think we are putting a set in the context, not a list.

Reply:
@kfaraz Yes, the problem is that `Set` gets serialized as `[x,y,z]`, which we aren't able to typecast into `Set` at this point. It runs into:
```
java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.util.Set
```

Hence had to typecast it as List at this layer of the code flow.